### ISSUE TITLE
Pkg.test() with Julia-0.5 fails

### ISSUE THREAD

**Issue by mdpradeep** [JuliaStats/Distributions.jl#542]
```
julia> Pkg.test("Distributions")
INFO: Computing test dependencies for Distributions...
INFO: Upgrading JSON: v0.7.0 => v0.7.1
INFO: Testing Distributions
Running tests:
        From worker 5:      testing Distributions.CategoricalDirectSampler
        From worker 5:      testing Distributions.AliasTable
        From worker 5:      testing Distributions.BinomialGeomSampler
        From worker 5:      testing Distributions.BinomialTPESampler
        From worker 3:      testing Distributions.Categorical(K=2, p=[0.5,0.5])
        From worker 5:      testing Distributions.BinomialPolySampler
        From worker 3:      testing Distributions.Categorical(K=4, p=[0.1,0.3,0.
2,0.4])
        From worker 3:      testing Distributions.Categorical(K=3, p=[0.15,0.25,
0.6])
        From worker 5:      testing Distributions.BinomialAliasSampler
        From worker 5:      testing Distributions.PoissonCountSampler
        From worker 5:      testing Distributions.PoissonADSampler
        From worker 5:      testing Distributions.PoissBinAliasSampler
        From worker 5:      testing Distributions.ExponentialSampler
        From worker 5:      testing Distributions.GammaGDSampler
        From worker 5:      testing Distributions.GammaMTSampler
        From worker 5:      testing Distributions.GammaGSSampler
        From worker 5:      testing Distributions.GammaIPSampler{S<:Distribution
s.Sampleable{Distributions.Univariate,Distributions.Continuous}}
ERROR: LoadError: On worker 2:
LoadError: can only precompile from node 1
 in compilecache at .\loading.jl:560
 in require at .\loading.jl:393
 in include_string at .\loading.jl:441
 in include_from_node1 at .\loading.jl:491
 in #5 at C:\Users\julia\AppData\Local\JuliaPro-0.5.0.1\pkgs-0.5.0.1\v0.5\Distri
butions\test\runtests.jl:44
 in #627 at .\multi.jl:1421
 in run_work_thunk at .\multi.jl:1001
 in macro expansion at .\multi.jl:1421 [inlined]
 in #626 at .\event.jl:68
while loading C:\Users\julia\AppData\Local\JuliaPro-0.5.0.1\pkgs-0.5.0.1\v0.5\Di
stributions\test\univariates.jl, in expression starting on line 4
 in #remotecall_fetch#608(::Array{Any,1}, ::Function, ::Function, ::Base.Worker,
 ::String, ::Vararg{String,N}) at .\multi.jl:1070
 in remotecall_fetch(::Function, ::Base.Worker, ::String, ::Vararg{String,N}) at
 .\multi.jl:1062
 in #remotecall_fetch#611(::Array{Any,1}, ::Function, ::Function, ::Int64, ::Str
ing, ::Vararg{String,N}) at .\multi.jl:1080
 in remotecall_fetch(::Function, ::Int64, ::String, ::Vararg{Any,N}) at .\multi.
jl:1080
 in #remotecall_pool#691(::Array{Any,1}, ::Function, ::Function, ::Function, ::W
orkerPool, ::String, ::Vararg{String,N}) at .\workerpool.jl:93
 in remotecall_pool(::Function, ::Function, ::WorkerPool, ::String, ::Vararg{Str
ing,N}) at .\workerpool.jl:91
 in #remotecall_fetch#694(::Array{Any,1}, ::Function, ::Function, ::WorkerPool,
::String, ::Vararg{String,N}) at .\workerpool.jl:124
 in remotecall_fetch(::Function, ::WorkerPool, ::String, ::Vararg{String,N}) at
.\workerpool.jl:124
 in (::Base.###699#700#702{WorkerPool,##5#6})(::Array{Any,1}, ::Function, ::Stri
ng, ::Vararg{String,N}) at .\workerpool.jl:151
 in (::Base.##699#701)(::String, ::Vararg{String,N}) at .\workerpool.jl:151
 in macro expansion at .\asyncmap.jl:63 [inlined]
 in (::Base.##757#759{Base.AsyncCollector,Base.AsyncCollectorState})() at .\task
.jl:360
 in sync_end() at .\task.jl:311
 in done(::Base.AsyncCollector, ::Base.AsyncCollectorState) at .\asyncmap.jl:124

 in pump_source(::Base.AsyncGenerator, ::Base.AsyncGeneratorState) at .\asyncmap
.jl:185
 in next(::Base.AsyncGenerator, ::Base.AsyncGeneratorState) at .\asyncmap.jl:201

 in _collect(::UnitRange{Int64}, ::Base.AsyncGenerator, ::Base.HasEltype, ::Base
.SizeUnknown) at .\array.jl:282
 in #pmap#716(::Bool, ::Int64, ::Void, ::Int64, ::Float64, ::Function, ::Void, :
:Void, ::Void, ::Base.#pmap, ::WorkerPool, ::##5#6, ::Array{String,1}) at .\pmap
.jl:121
 in pmap(::WorkerPool, ::Function, ::Array{String,1}) at .\pmap.jl:80
 in #pmap#728(::Array{Any,1}, ::Function, ::Function, ::Array{String,1}) at .\pm
ap.jl:146
 in pmap(::Function, ::Array{String,1}) at .\pmap.jl:146
 in include_from_node1(::String) at .\loading.jl:488
 in process_options(::Base.JLOptions) at .\client.jl:262
 in _start() at .\client.jl:318
while loading C:\Users\julia\AppData\Local\JuliaPro-0.5.0.1\pkgs-0.5.0.1\v0.5\Di
stributions\test\runtests.jl, in expression starting on line 43
============================[ ERROR: Distributions ]============================


failed process: Process(`'C:\Users\julia\AppData\Local\JuliaPro-0.5.0.1\Julia-0.
5.0\bin\julia' -Cx86-64 '-JC:\Users\julia\AppData\Local\JuliaPro-0.5.0.1\Julia-0
.5.0\lib\julia\sys.dll' --compile=yes --depwarn=yes --check-bounds=yes --code-co
verage=none --color=yes --compilecache=yes 'C:\Users\julia\AppData\Local\JuliaPr
o-0.5.0.1\pkgs-0.5.0.1\v0.5\Distributions\test\runtests.jl'`, ProcessExited(1))
[1]

================================================================================

INFO: Downgrading JSON: v0.7.1 => v0.7.0
ERROR: Distributions had test errors
 in #test#61(::Bool, ::Function, ::Array{AbstractString,1}) at .\pkg\entry.jl:74
0
 in (::Base.Pkg.Entry.#kw##test)(::Array{Any,1}, ::Base.Pkg.Entry.#test, ::Array
{AbstractString,1}) at .\<missing>:0
 in (::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#test,Tuple{Array{Abstract
String,1}}})() at .\pkg\dir.jl:31
 in cd(::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#test,Tuple{Array{Abstra
ctString,1}}}, ::String) at .\file.jl:48
 in #cd#1(::Array{Any,1}, ::Function, ::Function, ::Array{AbstractString,1}, ::V
ararg{Array{AbstractString,1},N}) at .\pkg\dir.jl:31
 in (::Base.Pkg.Dir.#kw##cd)(::Array{Any,1}, ::Base.Pkg.Dir.#cd, ::Function, ::A
rray{AbstractString,1}, ::Vararg{Array{AbstractString,1},N}) at .\<missing>:0
 in #test#3(::Bool, ::Function, ::String, ::Vararg{String,N}) at .\pkg\pkg.jl:25
8
 in test(::String, ::Vararg{String,N}) at .\pkg\pkg.jl:258
```

**Comment by jmxpearson** [comment]
Duplicate of #538.

**Comment by jingpengwu** [comment]
In real-world big projects, `precompile before parallel processing` may reduce the ease of usage. Would be nice to enable distributed compilation.

**Comment by ararslan** [comment]
Fixed by https://github.com/JuliaStats/Distributions.jl/pull/539

**Comment by ararslan** [comment]
@jingpengwu I'm not sure I agree. The problem with that is that if precompilation fails for whatever reason on one of the nodes, things can turn into a mess. If you precompile once and reuse the compiled cache everywhere, that's one less thing that you have to worry about, since you'll know that compilation on the head node either worked, in which case you can proceed safely, or precompilation failed and the program will exit before even doing anything in parallel.
